xija/runner.py

```diff
@@ -0,0 +1,212 @@
+
+        This could in theory be a tri-state
+        value: OK, not OK, and almost OK (meaning that the worst case violation
+        was within e.g. 20% of the model error).
+
+        :returns: bool (or str?)
+        """
+        pass
+
+
+class AcaModelRunner(ModelRunner):
+    """
+    Specific model components required to run the ACA model.
+
+    Note the limit and model_err values are defaults and could be set
+    directly (from ORviewer) with something like
+    ``AcaModelRunner.aacccdpt.limit = -7.9``.
+    """
+    # These class attributes represent the initial state and required data
+    # values to run the model.
+
+    pitch = ModelComp(float)
+    eclipse = ModelComp(bool)
+    aacccdpt = ModelComp(float, limit=-8.5, model_err=1.0)  # Node
+    aca0 = ModelComp(float)  # Pseudo-node
+
+
+class ObsReqRunner:
+    def __init__(self, ephemeris, eclipse, model_names=DEFAULT_MODELS):
+        """
+        :param ephemeris: dict of ephemeris {time (cxcsec), x, y, z}
+        """
+        self.ephemeris = ephemeris  # needs validation and processing
+        self.obs_reqs = {}
+        self.models = []
+
+        self.model_names = model_names.copy()
+
+        for model_name in model_names:
+            model_runner_name = model_name.capitalize() + 'ModelRunner'
+            self.models[model_name] = globals()[model_runner_name]()
+
+    def add_obs_req(self, obsid, att_targ, offset_targ_y, offset_targ_z,
+                    off_nom_roll,  # ??
+                    detector, n_fep, sim_z, duration):
+        """
+        Add an OR observation to the set of available observations.
+
+        This is a one-time operation before the first call to ``run()``.  This
+        also does the conversion of ``att
+        """
+        att_aca = calc_aca_from_targ(att_targ, offset_targ_y, offset_targ_z)
+
+        self.obs_reqs[obsid] = {'att_aca': att_aca,
+                                'att_targ': att_targ}
+        # etc
+
+    def set_initial_state(self, comp_names, comp_values):
+        """Supply initial state values for every component name in the
+        set of models in the ObsReqRunner object.
+
+        This represents the complete state at the end of the currently planned
+        schedule, effectively the continuity.
+
+        Examples of names that need to be provided are: pitch, aoattqt<N>, ccd_count,
+        fep_count, aaccdpt (node value), 1cbat, sim_px (pseudo-node values), sim_z.
+
+        This does assume that comp_names are unique, i.e. there are no models
+        that use the same component name for two different values.  That is
+        currently OK, and should not be a problem in the future.
+
+        IMPLEMENTATION NOTE: can comp_values from MATLAB be a list including
+        different data types?  If not then this method needs to operate on
+        one comp_name/value pair at once.  This is conceptually nicer but
+        probably a bit slower.
```

Comment:
Heterogeneous data can be passed into a python function call from MATLAB using pyexec, although, it may still make more sense to transfer the data into individual variables and then call construct a list/dict/whatever data structure will be used as the inputs in python itself.